pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -375,7 +375,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 func updateTaskRunsStatus(pr *v1alpha1.PipelineRun, pipelineState []*resources.ResolvedPipelineRunTask) {
 	for _, rprt := range pipelineState {
 		if rprt.TaskRun != nil {
-			pr.Status.TaskRuns[rprt.TaskRun.Name] = rprt.TaskRun.Status
+			prtrs := pr.Status.TaskRuns[rprt.TaskRun.Name]
+			if prtrs == nil {
+				prtrs = &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: fmt.Sprintf("%s-%s", pr.Name, rprt.PipelineTask.Name),
+				}
+			}
+			prtrs.Status = &rprt.TaskRun.Status
+			pr.Status.TaskRuns[rprt.TaskRun.Name] = prtrs
```

Comment:
i think this assignment only needs to happen in the `if prtrs == nil` case, since if it's not nil we're already holding a pointer to the thing we're changing

Reply:
Okiedokie.